Snippets/Snippets_5/Features/FeatureConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace Snippets5.Features
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+
+    class MyMagicFeature : Feature
+    {
+
+#region FeatureConfiguration
+        public MyMagicFeature()
+        {
+            EnableByDefault();
+
+            DependsOn<Sagas>();
+
+            Prerequisite(c => c.Settings.EndpointName().StartsWith("Magical"), "This feature only works on magical endpoints");
+        }
+#endregion
+
+#region FeatureSetup
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            // retrieve some global settings
+            var supervised = context.Settings.Get<bool>("SupervisorActive");
+
+            if (!supervised)
+            {
+                // register a more performant cleanup process with the container
+                context.Container.ConfigureComponent<UnsafeCleanupComponent>(DependencyLifecycle.InstancePerCall);
+            }
+
+            // add a custom behavior to the message processing pipeline which does our cleanup.
+            context.Pipeline.Register<CleanupBehaviorRegistration>();
+        }
+#endregion
+
+        void ConfigureFeatures()
+        {
+#region EnableDisableFeatures
+            var configuration = new BusConfiguration();
+            
+            // enable Saga feature since our custom feature relies on it
+            configuration.EnableFeature<Sagas>();
+
+            // this is not required if the feature uses EnableByDefault()
+            configuration.EnableFeature<MyMagicFeature>();
```

Comment:
actually, that snipped should represent the endpoint config code and not the feature. I just wrote it into the same class. Does it suffice to make that more clear in the markdown file or is there a real value in extracting that method in a new class & file?